v7.c

```diff
@@ -9763,9 +9823,23 @@ static val_t i_eval_expr(struct v7 *v7, struct ast *a, ast_off_t *pos,
       }
     case AST_CALL:
       return i_eval_call(v7, a, pos, scope);
+    case AST_VAR:
+      end = ast_get_skip(a, *pos, AST_END_SKIP);
+      ast_move_to_children(a, pos);
+      while (*pos < end) {
+        size_t name_len;
+        const char *name;
+        enum ast_tag tag = ast_fetch_tag(a, pos);
```

Comment:
Done.
Aslo, I've moved all ast_fetch_tag(a, pos) calls out of assert() invocation (with -DNDEBUG, those  important ast_fetch_tag(a, pos)  which shift position, will be optimized out).